expression/builtin_arithmetic_vec.go

```diff
@@ -164,11 +161,16 @@ func (b *builtinArithmeticModRealSig) vecEvalReal(input *chunk.Chunk, result *ch
 	if err := b.args[1].VecEvalReal(b.ctx, input, buf); err != nil {
 		return err
 	}
-
-	result.MergeNulls(buf)
+	if err := b.args[0].VecEvalReal(b.ctx, input, result); err != nil {
+		return err
+	}
 	x := result.Float64s()
 	y := buf.Float64s()
 	for i := 0; i < n; i++ {
+		if buf.IsNull(i) {
+			result.SetNull(i, true)
```

Comment:
We can use `result.MergeNulls(buf)` here.

Reply:
In some specific conditions, using `result.MergeNulls(buf)` won't pass unit test because of error handling problems. this line `c.Assert(2*vecWarnCnt, Equals, totalWarns)` won't pass. I have dicussed with @js00070 and confirmed it.

Reply:
Okay.
And did you join our vec-expr working group on Slack? 
I can't search your name in members of this working group.

Reply:
vectorized function should handle errors exactly the same as the non-vectorized function or the `vecWarnCnt` won't equal to the non-vec warn count. @qw4990

Reply:
> Okay.
> And did you join our vec-expr working group on Slack?
> I can't search your name in members of this working group.

I have joined vec-expr slack group now. and actually @js00070 and me are roommates, I can discuss with him offline.